src/main/java/org/snomed/snowstorm/ecl/validation/ECLEdgeCaseHandlerService.java

```diff
@@ -0,0 +1,131 @@
+package org.snomed.snowstorm.ecl.validation;
+
+import com.google.common.base.Functions;
+import io.kaicode.elasticvc.api.CommitListener;
+import io.kaicode.elasticvc.api.VersionControlHelper;
+import io.kaicode.elasticvc.domain.Commit;
+import org.snomed.langauges.ecl.ECLQueryBuilder;
+import org.snomed.langauges.ecl.domain.refinement.SubAttributeSet;
+import org.snomed.snowstorm.ecl.ECLQueryService;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SRefinedExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.expressionconstraint.SSubExpressionConstraint;
+import org.snomed.snowstorm.ecl.domain.refinement.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.snomed.snowstorm.core.data.domain.Concepts.CONCEPT_MODEL_DATA_ATTRIBUTE;
+import static org.snomed.snowstorm.ecl.domain.expressionconstraint.SExpressionConstraintHelper.MISSING;
+
+@Service
+public class ECLEdgeCaseHandlerService implements CommitListener {
```

Comment:
Running every query through a service named "edge case" seems counter intuitive. Maybe naming like `ECLPreprocessingService` would show the intent more clearly?